tests/forms_tests/tests/test_forms.py

```diff
@@ -3576,3 +3576,17 @@ class DataForm(Form):
         f = DataForm({'data': 'xyzzy'})
         self.assertTrue(f.is_valid())
         self.assertEqual(f.cleaned_data, {'data': 'xyzzy'})
+
+    def test_choicefield_uses_separate_widget_instances(self):
```

Comment:
Since this is testing a change in `Field`, it might be better in `forms_tests/field_tests/test_base.py` and named something like `test_field_deepcopies_widgets`.